### PR TITLE
New idl for testing setting the MemberId [20109]

### PR DESCRIPTION
**Pull request by richiware**

### CODE REVIEW

File: IDL/appendable.idl [hunk 1]
```diff
@@ -97,3 +97,8 @@ struct AppendableInheritanceStruct : AppendableShortStruct
 struct AppendableInheritanceEmptyStruct : AppendableShortStruct
 {
 };
+
+struct AppendableExtensibilityInheritance : AppendableShortStruct
+{
+    long var_long;
+};
```
Comment (JLBuenoLopez) [inline IDL/appendable.idl:104]: Should not the extensibility be explicit to avoid confusion? I suggest adding the extensibility annotation to every structure in the test IDLs. It might be done in another PR too.

File: IDL/final.idl [hunk 2]
```diff
@@ -97,3 +97,8 @@ struct FinalInheritanceStruct : FinalShortStruct
 struct InheritanceEmptyStruct : FinalShortStruct
 {
 };
+
+struct FinalExtensibilityInheritance : FinalShortStruct
+{
+    long var_long;
+};
```
Comment (JLBuenoLopez) [inline IDL/final.idl:104]: Is this structure meant to be appendable or final? As I have previously suggested, I think explicitly stating the extensibility is for the best.

It might be interesting to keep this structure but also define the final one following the specification XTypes v1.3 clause 7.2.2.4.5

> A Structure Type may inherit from another Structure Type as long as the following conditions are met:
>
> * The derived structure either has the extensibility kind unspecified, or else, it has the same extensibility kind as the base type.

This suggestion applies to the structures added into the other extensibility IDLs.

File: IDL/key.idl [hunk 3]
```diff
@@ -83,13 +83,13 @@ struct KeyedEmptyStruct
 
 struct KeyedEmptyInheritanceStruct : KeyedEmptyStruct
 {
-    @key string key_str;
+    string key_str;
```
Comment (JLBuenoLopez) [inline IDL/key.idl:86]: Great! And nice to include the check in IDL Parser!

File: IDL/member_id.idl [hunk 4]
```diff
@@ -0,0 +1,128 @@
+@mutable
+struct FixId
+{
+    @id(100)
+    octet o;
```
Comment (JLBuenoLopez) [inline IDL/member_id.idl]: Personally, I think it is easier to read if the annotation and the type are in the same line as it is usually written in the specifications. Otherwise, it might be helpful to include a blank line between types. Though this is only a style suggestion and as there is no IDL code style, it can be overlooked.

File: IDL/member_id.idl [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+{
+    char cd;
+    @id(101)
+    octet od;
+    @hashid
+    short sd;
+    @hashid("long2")
+    long ld;
+    long long lld;
+};
```
Comment (JLBuenoLopez) [inline IDL/member_id.idl]: There is a specific inheritance test case that it would be nice to test as well. What happens if there is nested inheritance as the following IDL:

```
struct SomeStruct
{
    short some_var;
};

struct EmptyStruct : SomeStruct
{
};

struct AnotherInheritedStruct : EmptyStruct
{
    long another_var;
};
```

Probably the logic in IDL Parser would handle this case correctly, but I suggest having the test case to ensure that in the future is not broken.

**Review by JLBuenoLopez — CHANGES_REQUESTED** [review]

File: IDL/appendable.idl (hunk 1 above)
Comment (richiware) [reply IDL/appendable.idl:104]: The case you are talking about is already tested. This case tests the inheritance when a derived structure 
does not have the extensibility defined but the base structure does.

File: IDL/final.idl (hunk 2 above)
Comment (richiware) [reply IDL/final.idl:104]: The case you are talking about is already tested. This case tests the inheritance when a derived structure 
does not have the extensibility defined but the base structure does.

File: IDL/member_id.idl (hunk 5 above)
Comment (richiware) [reply IDL/member_id.idl]: Done in 65376cd

File: IDL/member_id.idl (hunk 4 above)
Comment (richiware) [reply IDL/member_id.idl]: Done in 65376cd

**Review by JLBuenoLopez — APPROVED** [review]
LGTM
